lib/matplotlib/axis.py

```diff
@@ -625,6 +626,40 @@ class Ticker(object):
     formatter = None
 
 
+class _LazyDefaultTickList(object):
+    """
+    A lazy evaluating placeholder for the default tick list.
+
+    On access the class replaces itself with the default 1-element list. We
+    use the lazy evaluation so that :meth:`matplotlib.axis.Axis.reset_ticks`
+    can be called multiple times without the overhead of initialzing every
+    time.
+
+    https://github.com/matplotlib/matplotlib/issues/6664
```

Comment:
I tend to consider that checking the git blame is sufficient to see where this comes from, and prefer not adding explicit links to issues (otherwise it's a bit endless).  Not a huge deal of course.